phac-nml/irida-next
# Sample attachments

## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR adds in support for attaching files to Samples. Currently this only allows a User with `role >= Maintainer` to attach files to a Sample.

This PR also adds an id to the layout component div for flashes so that `turbo_stream` requests can display flashes. This is used in the `create` and `destroy` actions.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

![image](https://github.com/phac-nml/irida-next/assets/492127/321b7de6-6619-44ec-978a-0885c09f964a)

![image](https://github.com/phac-nml/irida-next/assets/492127/ee1c87d3-d656-4428-b42d-3b2c2e5d99de)

![image](https://github.com/phac-nml/irida-next/assets/492127/9f682b34-1429-4394-b9be-0b842f49244b)


![image](https://github.com/phac-nml/irida-next/assets/492127/f14f5b7c-d9b3-4565-b90f-db75fab1411e)

![image](https://github.com/phac-nml/irida-next/assets/492127/9a998626-4ef4-4215-a402-ea3acce732ea)

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Migrate your DB with `bin/rails db:migrate`
2. Navigate to a Sample and click `Upload File`
3. Select a file and click `Upload File`
4. Confirm that it appears in the table
5. Repeat this process and confirm that a new file appears in the table and the existing file remains.
6. Click "Delete" on one of the files you uploaded, verify that the confirm dialog appears, and verify that it is removed from the table after clicking confirm.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
